python/ray/_private/metrics.py

```diff
@@ -0,0 +1,67 @@
+import functools
+import time
+from ray.util.metrics import Counter, Histogram
+from typing import Dict, Callable
+
+
+class FunctionMetricsCollector:
+    """A helper class that implements the context manager and decorator
+    for recording function/block execution.
```

Comment:
Can you add usage docstring here? E.g.,https://github.com/ray-project/ray/blob/414737b7c71f4f4b4e1e062ccde4b654db5538a1/python/ray/serve/api.py#L532